### PR TITLE
Fix nested accordion keyboard navigation

### PR DESCRIPTION
**Pull request by ryuran**
- Fix navigation #27
- Replace jscs & jshint by eslint
- Add travis file config for Continuous Integration (ask for help to plug travis to github if needed)
- Remove legacy file (because legacy is for git history)
- Add npmignore file

### CODE REVIEW

**Comment by nico3333fr** [comment]
Hello,

I have a look at this this week-end, thanks a lot ! 👍

File: bower.json [hunk 1]
```diff
@@ -26,7 +26,5 @@
   ],
   "dependencies": {
     "jquery": ">=1.0.0"
-  },
-  "devDependencies": {
```
Comment (ryuran) [inline bower.json:30]: I remove dev dependencies from bower because it should not be used for dev on this project.

File: bower.json [hunk 2]
```diff
@@ -1,8 +1,8 @@
 {
   "title": "jQuery Accessible Accordion system, using ARIA",
   "name": "jquery-accessible-accordion-aria",
-  "version": "2.4.4",
-  "main": "jquery-accessible-accordion-aria.js",
+  "version": "2.4.5",
+  "main": "index.js",
```
Comment (ryuran) [inline bower.json:5]: I rename main file to `index.js` to keep a consistant standard cross project.

File: docs/sample-nested.html [hunk 3]
```diff
@@ -0,0 +1,101 @@
+<!DOCTYPE html>
```
Comment (ryuran) [inline docs/sample-nested.html:1]: I add this page to test nested accordion.

File: docs/sample.html [hunk 4]
```diff
@@ -28,7 +27,7 @@
 }
 
 button.js-accordion__header {
-  display: block;
+  display: block !important;
```
Comment (ryuran) [inline docs/sample.html:30]: selector was not strong enough, we should clean it.

File: index.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 /*
  * jQuery Accessible Accordion system, using ARIA
- * @version v2.4.4 
+ * @version v2.4.4
```
Comment (ryuran) [inline index.js:3]: I forgot to up version here.

File: index.js [hunk 6]
```diff
@@ -11,7 +11,7 @@
     } else if (typeof exports !== 'undefined') {
         module.exports = factory(require('jquery'));
     } else {
-        factory(jQuery);
+        factory(window.jQuery);
```
Comment (ryuran) [inline index.js:14]: To not use Global.

File: index.js [hunk 7]
```diff
@@ -52,14 +52,10 @@
         }).addClass(this.options.prefixClass);
 
         // id generated if not present
-        this.$wrapper.each($.proxy(function(index, el) {
```
Comment (ryuran) [inline index.js:55]: `this.$wrapper` is alone. `.each(` is useless here.

File: index.js [hunk 8]
```diff
@@ -169,8 +165,7 @@
         var $button = $target.is('button') ? $target : $target.closest('button');
         var $firstButton = this.$buttons.first();
         var $lastButton = this.$buttons.last();
-        var $prevButton = $button.prevAll(this.options.buttonsSelector).first();
```
Comment (ryuran) [inline index.js:172]: The problem was here.
`$button.prevAll(this.options.buttonsSelector)` with a relative `this.options.buttonsSelector`by exemple `> button` can’t work.
So I change solution using index to know current, last and first position.

File: index.js [hunk 9]
```diff
@@ -224,10 +219,9 @@
     Accordion.prototype.keydownPanelEventHandler = function(e) {
         var $panel = $(e.target).closest(this.options.panelsSelector);
         var $button = $('#' + $panel.attr('aria-labelledby'));
-        var $firstButton = this.$wrapper.find(this.options.buttonsSelector).first();
```
Comment (ryuran) [inline index.js:227]: Same here

File: jquery-accessible-accordion-aria-legacy-1.2.0.js [hunk 10]
```diff
@@ -1,225 +0,0 @@
-$(document).ready(function(){
```
Comment (ryuran) [inline jquery-accessible-accordion-aria-legacy-1.2.0.js:1]: This old code shouldn't stay in master branch

File: package.json [hunk 11]
```diff
@@ -2,39 +2,45 @@
   "title": "jQuery Accessible Accordion system, using ARIA",
   "name": "jquery-accessible-accordion-aria",
   "description": "This jQuery plugin will provide you an accessible Accordion System, using ARIA",
```
Comment (ryuran) [inline package.json:4]: I move information to top of file and dependencies to bottom

**Comment by nico3333fr** [comment]
Well, I had time to test all of this... and everything seems perfect.

I merge 👍 

You're great ;)

**Comment by nico3333fr** [comment]
Just a comment:

``````

**Pull request merged**